src/player.ts

```diff
@@ -62,6 +63,30 @@ export default class Player {
     return factions.planet(this.faction);
   }
 
+  payCosts(costs: Reward[]) {
+    for (let cost of costs) {
+      this.data.payCost(this.factionReward(cost));
+    }
+  }
+
+  gainRewards(rewards: Reward[]) {
+    for (let reward of rewards) {
+      this.data.gainReward(this.factionReward(reward));
+    }
+  }
+
+
+  canPay(reward: Reward[]): boolean {
+    const rewards = Reward.merge(reward);
+
+    for (const reward of rewards) {
+      if (!this.data.hasResource(this.factionReward(reward))) {
```

Comment:
This is wrong. 

> To make a Gaia Planet habitable, pay one ore instead of one Q.I.C.

This is the only time gleens pay ores instead of Q.I.C. And from what I understand, they still do even if their academy is built.

Reply:
You are right. They get ores instead of qics until academy2. They always pay 2ores for a gaia.
So the only place where factionReward is need is in GainRewards, assuming I manage gleens' gaia cost in build().
I'm in the middle of the freeActions after an action (e.g build). Wait and I'll post everything together.